segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,215 @@
+        self.coo_cols = coo_cols
+        self.coo_data = coo_data
+        self.image_shape = image_shape
+        self.segmentation_id = segmentation_id
+        self.roi_id = roi_id
+        self._sparse_coo = None
+        self._dense_mat = None
+        self._edges = None
+
+    @classmethod
+    def roi_from_query(cls, segmentation_id:int,
+                       roi_id: int) -> "ROI":
+        """
+        Queries and builds ROI object by querying LIMS table for
+        produced labeling ROIs.
+        Args:
+            segmentation_id: Id of the segmentation run
+            roi_id: Unique Id of the ROI to be loaded
+
+        Returns: ROI object for the given segmentation_id and roi_id
+        """
+        label_vars = query_utils.get_labeling_env_vars()
+
+        shape = query_utils.query(
+            f"SELECT * FROM public.segmentation_runs WHERE id={segmentation_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)[0]['video_shape']
+
+        roi = query_utils.query(
+            f"SELECT * FROM public.rois WHERE "
+            f"segmentation_run_id={segmentation_id} AND id={roi_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)
+        return ROI(coo_rows=roi['coo_rows'][0],
+                   coo_cols=roi['coo_cols'][0],
+                   coo_data=roi['coo_data'][0],
+                   image_shape=shape,
+                   segmentation_id=segmentation_id,
+                   roi_id=roi_id)
+
+    @property
+    def sparse_coo(self):
+        """
+        Produces a sparse coo matrix scipy object from the
+        Returns: A sparse_csc scipy object for the data
+        """
+        if self._sparse_coo is None:
+            self._sparse_coo = coo_matrix((
+                self.coo_data,
+                (self.coo_rows,
+                 self.coo_cols)),
+                shape=self.image_shape)
+        return self._sparse_coo
+
+    @property
+    def dense_matrix(self):
+        """
+        Produces dense matrix from spare_coo
+        Returns: Dense matrix for ROI
+        """
+        if self._dense_mat is None:
+            self._dense_mat = self.sparse_coo.toarray()
+        return self._dense_mat
+
+    def generate_binary_mask_from_threshold(self, threshold: float):
+        """
+        Simple binary mask from a provided threshold.
+        Args:
+            threshold: The threshold to compare values
+
+        Returns:
+
+        Notes:
+            This function will likely become deprecated as we work on more
+            applicable binary thresh holding
+        """
+        return np.where(self.dense_matrix >= threshold, 1, 0)
+
+    def generate_roi_inner_edge_coordinates(self, stroke_size: int,
```

Comment:
I think you can do a lot of this with opencv. Did you check out `contours`?
https://docs.opencv.org/3.4/d4/d73/tutorial_py_contours_begin.html

Also, I noticed opencv was a requirement, but here you are using ndimage heavily. Using multiple image-processing libraries tends to lead to a lot of conversions and confusion. If you can simply base on ndimage OR opencv, it ends up being easier in the end.